addons/crm/tests/test_crm_lead.py

```diff
@@ -246,16 +246,18 @@ def test_crm_lead_create_pipe_data(self):
 
     @users('user_sales_manager')
     def test_crm_lead_currency_sync(self):
-        self.company_main.currency_id = self.env.ref('base.EUR')
-
-        lead = self.env['crm.lead'].create({
+        lead_company = self.env['res.company'].sudo().create({
```

Comment:
This change is globally unrelated. Why woudln't change the currency, why do we have to create a new company, why here and not somewhere else ?

Reply:
> why do we have to create a new company, why here and not somewhere else ?

All accounting tests are creating a new company and a new currency... It has been like since at least 13.0

Reply:
But you are not in accounting here. What is the real issue when installing CRM ?

Reply:
I didn't write this code, and I admit I'm not sure it is useful anymore. Removed, we'll see if it was still needed

Reply:
Ok, thanks :)

Reply:
```
odoo.addons.crm.tests.test_crm_lead:249

ERROR: Subtest TestCRMLead.test_crm_lead_currency_sync (login='user_sales_manager')
Traceback (most recent call last):
  File "<decorator-gen-1052>", line 2, in test_crm_lead_currency_sync
  File "/data/build/odoo/odoo/tests/common.py", line 2025, in _users
    func(*args, **kwargs)
  File "/data/build/odoo/addons/crm/tests/test_crm_lead.py", line 249, in test_crm_lead_currency_sync
    self.company_main.currency_id = self.env.ref('base.EUR')
  File "/data/build/odoo/odoo/fields.py", line 1303, in __set__
    records.write({self.name: write_value})
  File "/data/build/odoo/addons/account/models/company.py", line 366, in write
    raise UserError(_('You cannot change the currency of the company since some journal items already exist'))
odoo.exceptions.UserError: You cannot change the currency of the company since some journal items already exist
                                  
```

Reply:
Is this a better way of fixing?
```diff
diff --git a/addons/crm/tests/test_crm_lead.py b/addons/crm/tests/test_crm_lead.py
index e40059c4ead..9247c1fa3a7 100644
--- a/addons/crm/tests/test_crm_lead.py
+++ b/addons/crm/tests/test_crm_lead.py
@@ -246,7 +246,8 @@ class TestCRMLead(TestCrmCommon):
 
     @users('user_sales_manager')
     def test_crm_lead_currency_sync(self):
-        self.company_main.currency_id = self.env.ref('base.EUR')
+        from odoo.models import Model
+        Model.write(self.company_main, {'currency_id': self.env.ref('base.EUR')})
 
         lead = self.env['crm.lead'].create({
             'name': 'Lead 1',
@@ -254,7 +255,7 @@ class TestCRMLead(TestCrmCommon):
         })
         self.assertEqual(lead.company_currency, self.env.ref('base.EUR'))
 
-        self.company_main.currency_id = self.env.ref('base.CHF')
+        Model.write(self.company_main, {'currency_id': self.env.ref('base.CHF')})
         lead.with_company(self.company_main).update({'company_id': False})
         self.assertEqual(lead.company_currency, self.env.ref('base.CHF'))
```

Reply:
Ok, it is a constraint raised when you have both modules... as it is globally a related not stored + fallback on current company, indeed it makes sense to create a new company to make the test more explicit and not dependent on accounting error anymore

Idea is to test:
  * no company -> env.company.currency
  * change mocpany -> update to this company

Sorry for back and forth then :) having another company and test the compute field is sufficient for this test.